docs/07-lifecycle.md

```diff
@@ -2,8 +2,68 @@
 ## Development
 ## Submit to Marketplace
 ## Provision
+### App Bundle
+Bundle is a convenient way to deliver an app to the Mattermost ecosystem. It provides a way to organize code and resources needed for an app to run. A bundle is created by the developer of the app. Mattermost uses bundles to provision and install/uninstall apps. 
```

Comment:
I suggest we go with `app bundle` as `bundle` is quite generic and it's not clear on first sight what it means.

```suggestion
An app bundle is a convenient way to deliver an app to the Mattermost ecosystem. It provides a way to organize code and resources needed for an app to run. An app bundle is created by the developer of the app. Mattermost uses app bundles to provision and install/uninstall apps. 
```